src/validation/validate.js

```diff
@@ -52,3 +52,39 @@ export function validate(
   visit(documentAST, visitWithTypeInfo(typeInfo, visitor));
   return context.getErrors();
 }
+
+export function validateSDL(
+  documentAST: DocumentNode,
+  schemaToExtend?: ?GraphQLSchema,
+  rules?: $ReadOnlyArray<SDLValidationRule> = specifiedSDLRules,
+): $ReadOnlyArray<GraphQLError> {
+  const context = new SDLValidationContext(documentAST, schemaToExtend);
+  const visitors = rules.map(rule => rule(context));
+  visit(documentAST, visitInParallel(visitors));
+  return context.getErrors();
+}
+
+/**
+ * Utility function which asserts a SDL document is valid by throwing an error
+ * if it is invalid.
+ */
+export function assertValidSDL(documentAST: DocumentNode): void {
```

Comment:
Should we make this API public yet? We don't have a corresponding `assertValidAST` yet, nor `assertValid` (for both SDL and AST). Internally, we do the exact same thing you're doing here internally, but we should probably offer all the functionality (maybe with a single `assertValid` function for now). I'm also not 100% sure the `assertValid` functions belongs here vs. a utility file, and I am not sure we should offer them without offering the ability to pass in your own set of rules.